TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/RoadNode.cs

```diff
@@ -151,6 +151,17 @@ public void UpdateIntersectionJunctionEdgeNavigation(Road road)
                 curr = curr.Next;
             }
         }
+        public float GetDistanceToEndRoadNode()
```

Comment:
This should just be called `GetDistanceToEnd`, since it is a method on RoadNodes it does not make sense to specify RoadNode in the same. Could also be a getter instead of a function if that seems desirable

Also, this should be a generic method on the `Node` class instead of just for RoadNodes